### ISSUE TITLE
Severe Performance Degression (Unification?)

### ISSUE THREAD

**Issue by theduke** [intellij-rust/intellij-rust#1769]
The recent release caused a very painful performance degression for me.

Tiny code changes will drive up CPU usage to 100% all the time, and much worse on waiting for autocomple or just during typing, the entire UI completely freezes for several seconds on a regular basis (like every few minutes).

This is pretty much unacceptable for me and enough to drop the plugin.

Since this is probably due to unification, could there be an option to disable it? (at least until performance optimizations can be made)

**Comment by matklad** [comment]
No, there's no way to disable new type inference, but you can always find previous releases of the plugin at http://plugins.jetbrains.com/plugin/8182-rust.

Could you share a project where you observe performance degradation? We would love to fix it! :)

**Comment by theduke** [comment]
I've tried with multiple projects and it's an issue with every crate I've tried, even small ones of only a few hundred lines.

The problems get worse the bigger the crate is.
